src/themes/styled.d.ts

```diff
@@ -0,0 +1,15 @@
+import 'styled-components'
```

Comment:
`src/themes/styled.d.ts`

This is a "d" file I've read about. I don't understand how they work or when we should create them though.

Reply:
OK, time to explain I guess. :smile: These files contains definitions of types for TS to use when working with not naturally typed code. It can be also used to modify or extends already existing types used in external packages (`node_module/`).

In this case, I used it to extend otherwise empty `DefaultTheme` type from styled-components to define our custom theme properties. Now our themes definition with this type will be check by TS if anything is missing or so... :grin:

More about this topic in the TS official documentation - https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html

But be warned thus this is kinda advanced topic. :warning: :sweat_smile: